src/main/java/org/influxdb/impl/InfluxDBResultMapper.java

```diff
@@ -75,31 +75,63 @@
    * @param queryResult the InfluxDB result object
    * @param clazz the Class that will be used to hold your measurement data
    * @param <T> the target type
+   *
    * @return a {@link List} of objects from the same Class passed as parameter and sorted on the
-   *         same order as received from InfluxDB.
+   * same order as received from InfluxDB.
+   *
    * @throws InfluxDBMapperException If {@link QueryResult} parameter contain errors,
-   *         <tt>clazz</tt> parameter is not annotated with &#64;Measurement or it was not
-   *         possible to define the values of your POJO (e.g. due to an unsupported field type).
+   * <tt>clazz</tt> parameter is not annotated with &#64;Measurement or it was not
+   * possible to define the values of your POJO (e.g. due to an unsupported field type).
    */
   public <T> List<T> toPOJO(final QueryResult queryResult, final Class<T> clazz) throws InfluxDBMapperException {
+    throwExceptionIfMissingAnnotation(clazz);
+    String measurementName = getMeasurementName(clazz);
+    return this.toPOJO(queryResult, clazz, measurementName);
+  }
+
+  /**
+   * <p>
+   * Process a {@link QueryResult} object returned by the InfluxDB client inspecting the internal
+   * data structure and creating the respective object instances based on the Class passed as
+   * parameter.
+   * </p>
+   *
+   * @param queryResult the InfluxDB result object
+   * @param clazz the Class that will be used to hold your measurement data
+   * @param <T> the target type
+   * @param measurementName name of the Measurement
+   *
+   * @return a {@link List} of objects from the same Class passed as parameter and sorted on the
+   * same order as received from InfluxDB.
+   *
+   * @throws InfluxDBMapperException If {@link QueryResult} parameter contain errors,
+   * <tt>clazz</tt> parameter is not annotated with &#64;Measurement or it was not
+   * possible to define the values of your POJO (e.g. due to an unsupported field type).
+   */
+  public <T> List<T> toPOJO(final QueryResult queryResult, final Class<T> clazz, final String measurementName)
+      throws InfluxDBMapperException {
     Objects.requireNonNull(queryResult, "queryResult");
     Objects.requireNonNull(clazz, "clazz");
 
-    throwExceptionIfMissingAnnotation(clazz);
     throwExceptionIfResultWithError(queryResult);
     cacheMeasurementClass(clazz);
 
     List<T> result = new LinkedList<T>();
-    String measurementName = getMeasurementName(clazz);
+
+    if (measurementName == null || measurementName.isEmpty())
```

Comment:
Please move these verification to the top of the method. IMO `Objects.requireNonNull(measurementName, "measurementName");` would be enough and the code should be resilient against an empty string.